dis_006_reliefweb_disasters/contents/src/__init__.py

```diff
@@ -55,263 +72,445 @@
+                            # binary 'True' or 'False' value and covert it to a string                            
                             item = str(entry['fields']['current'])
+                        # if we are fetching data for event_type_ids column
                         elif key == 'event_type_ids':
+                            # get secondary event ids from ids list
                             item = ids
+                        # if we are fetching data for event types column
                         elif key == 'event_types':
+                            # get secondary event names from names list
                             item = names
-                        elif key == 'url':
-                            item = entry['fields']['url']
-                        elif key == 'lon':
-                            item = country['location']['lon']
-                        elif key == 'lat':
-                            item = country['location']['lat']
+                        # if we are fetching data for event name column
+                        elif key=='event_name':
+                            # get the event name from 'name' feature
+                            item = entry['fields']['name']
+                        # if we are fetching data for coordinates
+                        elif key == 'lon' or 'lat':
+                            # get the lat, long information from location feature
+                            item = country['location'][key]
+                        # for all other columns, we can fetch the data from fields feature 
+                        # using our column name in Carto
+                        else:
+                            item = entry['fields'][key]
                     except KeyError:
+                        # if the column we are trying to retrieve doesn't exist in the source data, store None
                         item=None
+                    # store the retrieved item from this row to a list
                     row.append(item)
+                # add the list of values from this row to the list of new data
                 new_data.append(row)
-
+    # find the length (number of rows) of new_data            
     num_new = len(new_ids)
+    # if we have found new dates to process
     if num_new:
+        # insert new data into the carto table
         logging.info('Adding {} new records'.format(num_new))
-        cartosql.blockInsertRows(CARTO_TABLE, CARTO_SCHEMA.keys(), CARTO_SCHEMA.values(), new_data)
+        cartosql.blockInsertRows(CARTO_TABLE, CARTO_SCHEMA.keys(), CARTO_SCHEMA.values(), new_data
+                                    , user=CARTO_USER, key=CARTO_KEY)
     return(num_new)
 
 def processInteractions():
+    '''
+    Process and upload data for interaction table. This additional interaction table was created to 
+    display multiple ongoing disaster events in a country.
```

Comment:
```suggestion
    display the multiple ongoing disaster events in a country as a single row in the Carto table. For each geometry, only one row of data will be used on Resource Watch map interactions, so when there are multiple events for a given geometry, we need to compact them into a single row so that all the information will be shown on an interaction.
```